### ISSUE TITLE
why not use monorepo?

### ISSUE THREAD

**Issue by KMJ-007** [tscircuit/tscircuit#320]
yo

come across this, cool project

rather than maintaining different repo, you can maintain one single repo using mono repo and all the internal packages also

any reasons why not to go for mono repo?

seeing the codebase would love to contribute!

**Comment by seveibar** [comment]
Hey @KMJ-007, glad to have you here!

There are already over 100 repos inside of tscircuit, we create repos at api boundaries that are super testable and understandable. We also create repos for experiments that are likely to have a lower code quality standard. Repos mature as time goes on, and get things like CI and CODEOWNERS. Dependencies upgrades are automatically run against almost all repos, and ensure that an upgrade doesnt break the tests for that specific repo.

In short, we run tscircuit like a community ecosystem rather than a monolithic piece of software, and this helps develop more reliably, and run more experiments without degrading code quality
